sympy/stats/tests/test_continuous_rv.py

```diff
@@ -46,7 +47,13 @@ def test_single_normal():
             2**S.Half*exp(-(x - mu)**2/(2*sigma**2))/(2*pi**S.Half*sigma))
 
     assert P(X**2 < 1) == erf(2**S.Half/2)
-    assert quantile(Y)(x) == Intersection(S.Reals, FiniteSet(sqrt(2)*sigma*(sqrt(2)*mu/(2*sigma) + erfinv(2*x - 1))))
+    ans = quantile(Y)(x)
+    eq = ans.atoms(Eq).pop()
+    ans = factor_terms(ans.xreplace({eq: Eq(eq.lhs.simplify().factor(), 0)}
+        ).xreplace({eq.atoms(Dummy).pop(): y}))
+    assert ans == Complement(ConditionSet(y, Eq((-mu +
+        y)*(2*x + erf(sqrt(2)*(mu - y)/(2*sigma)) - 1),
+        0), S.Reals), FiniteSet(mu))
```

Comment:
Why has the answer here become more complicated?

Reply:
Because `solveset` bails out if it can't solve the expression as written:
```python
>>> e
-_x*erf(sqrt(2)*(_x - mu)/(2*sigma)) - _x + mu*erf(sqrt(2)*(_x - mu)/(2*sigma))
+ mu + 2*x*(_x - mu)
>>> solveset(_, _x)
ConditionSet(_x, Eq(-_x*erf(sqrt(2)*(_x - mu)/(2*sigma)) - _x + mu*erf(sqrt(2)*(
_x - mu)/(2*sigma)) + mu + 2*x*(_x - mu), 0), Complexes)
>>> solveset(e.factor(), _x)
FiniteSet(mu, sqrt(2)*sigma*(sqrt(2)*mu/sigma + 2*erfinv(2*x - 1))/2)
```

Reply:
This has been fixed. The equation represents an interesting case where `signsimp` helps a lot by doing a simple thing. And then `factor` tidies up the simplified expression and gives a product of two factors which can be easily solved:
```
>>> fprint(e)

-w*exp(-w**2 + w*y - y**2)*exp(w**2 - w*y + y**2)*erf(w - y) +
w*exp(-w**2 + w*y - y**2)*exp(w**2 - w*y + y**2) + 2*x*(-w + y) +
y*exp(-w**2 + w*y - y**2)*exp(w**2 - w*y + y**2)*erf(w - y) -
y*exp(-w**2 + w*y - y**2)*exp(w**2 - w*y + y**2)
>>> signsimp(e)
-w*erf(w - y) + w - 2*x*(w - y) + y*erf(w - y) - y
>>> factor(_)
(-w + y)*(2*x + erf(w - y) - 1)
```